public/app/features/explore/RichHistory/RichHistoryCard.test.tsx

```diff
@@ -56,93 +79,140 @@ const starredQueryWithComment: RichHistoryQuery<MockQuery> = {
+    it('should have title "Add comment" at comment icon, if no comment present', async () => {
+      setup();
+      const addComment = await screen.findByTitle('Add comment');
+      const editComment = await screen.queryByTitle('Edit comment');
+      expect(addComment).toBeVisible();
+      expect(editComment).toBeNull();
     });
-    it('should open update comment form when edit comment button clicked', () => {
-      const wrapper = setup({ query: starredQueryWithComment });
-      const editCommentButton = wrapper.find({ title: 'Edit comment' });
-      editCommentButton.simulate('click');
-      expect(wrapper.find({ 'aria-label': 'Update comment form' })).toHaveLength(1);
+    it('should open update comment form when edit comment button clicked', async () => {
+      setup({ query: starredQueryWithComment });
+      const editComment = await screen.findByTitle('Edit comment');
+      fireEvent.click(editComment);
+      const updateCommentForm = await screen.findByLabelText('Update comment form');
+      expect(updateCommentForm).toBeVisible();
     });
-    it('should close update comment form when escape key pressed', () => {
-      const wrapper = setup({ query: starredQueryWithComment });
-      const editCommentButton = wrapper.find({ title: 'Edit comment' });
-      editCommentButton.simulate('click');
-      wrapper.simulate('keydown', { key: 'Escape' });
-      expect(wrapper.find({ 'aria-label': 'Update comment form' })).toHaveLength(0);
+    it('should close update comment form when escape key pressed', async () => {
+      setup({ query: starredQueryWithComment });
+      const editComment = await screen.findByTitle('Edit comment');
+      fireEvent.click(editComment);
+      const updateCommentForm = await screen.findByLabelText('Update comment form');
+      fireEvent.keyDown(getByText(updateCommentForm || new HTMLElement(), starredQueryWithComment.comment), {
+        key: 'Escape',
+      });
+      const findCommentForm = screen.queryByLabelText('Update comment form');
+      expect(findCommentForm).toBeNull();
     });
-    it('should close update comment form when enter and shift keys pressed', () => {
-      const wrapper = setup({ query: starredQueryWithComment });
-      const editCommentButton = wrapper.find({ title: 'Edit comment' });
-      editCommentButton.simulate('click');
-      wrapper.simulate('keydown', { key: 'Enter', shiftKey: true });
-      expect(wrapper.find({ 'aria-label': 'Update comment form' })).toHaveLength(0);
+    it('should close update comment form when enter and shift keys pressed', async () => {
+      setup({ query: starredQueryWithComment });
+      const editComment = await screen.findByTitle('Edit comment');
+      fireEvent.click(editComment);
+      const updateCommentForm = await screen.findByLabelText('Update comment form');
+      fireEvent.keyDown(getByText(updateCommentForm || new HTMLElement(), starredQueryWithComment.comment), {
```

Comment:
Wonder why we need `new HTMLElement()` here?

Reply:
There must have been another issue this covered.. I was getting an error at some point but another change must have fixed it.